aten/src/ATen/native/cpu/BlasKernel.cpp

```diff
@@ -53,32 +53,30 @@ auto sum(int64_t N, Func f) {
   return partial_sums[0];
 }
 
-
 template <typename scalar_t, typename opmath_t>
-void gemm_notrans_(
-    int64_t m, int64_t n, int64_t k,
+void gemm_notrans_or_transb(
+    int64_t m,
+    int64_t n,
+    int64_t k,
     opmath_t alpha,
-    const scalar_t *a, int64_t lda,
-    const scalar_t *b, int64_t ldb,
+    const scalar_t* a,
+    int64_t lda,
+    const scalar_t* b,
+    int64_t ldb,
     opmath_t beta,
-    scalar_t *c, int64_t ldc) {
+    scalar_t* c,
+    int64_t ldc) {
   // c *= beta
   scale_(m, n, beta, c, ldc);
 
   // c += alpha * (a @ b)
-  for (const auto l : c10::irange(k)) {
+  for (const auto i : c10::irange(m)) {
     for (const auto j : c10::irange(n)) {
-      opmath_t val = b[l + j * ldb] * alpha;
-      int64_t i_m = m / 4;
-      for (const auto i_i : c10::irange(i_m)) {
-        c[j * ldc + i_i * 4 + 0] += a[i_i * 4 + 0 + l * lda] * val;
-        c[j * ldc + i_i * 4 + 1] += a[i_i * 4 + 1 + l * lda] * val;
-        c[j * ldc + i_i * 4 + 2] += a[i_i * 4 + 2 + l * lda] * val;
-        c[j * ldc + i_i * 4 + 3] += a[i_i * 4 + 3 + l * lda] * val;
-      }
-      int64_t i = i_m * 4;
-      for (; i < m; i++)
-        c[j * ldc + i] += a[i + l * lda] * val;
+      const auto dot = sum(k, [&](int64_t l) -> opmath_t {
+        return static_cast<opmath_t>(a[i * lda + l]) *
+            static_cast<opmath_t>(b[j * ldb + l]);
```

Comment:
What does this function assume about `b`'s transposition? Here we assume `b` is notrans, right?

Reply:
Yes, it is wrong here.
But I confirmed
```        
return static_cast<opmath_t>(a[l * lda + i]) *
            static_cast<opmath_t>(b[j * ldb + l]);
```
can work for whether b is transposed or not.
`transpose` will impact the value for `ldb` which will make  `b[j * ldb + l]` works anyway.

Reply:
> `transpose` will impact the value for `ldb` which will make `b[j * ldb + l]` works anyway.

It will be valid to read, but you'll be reading the wrong dimension.

Reply:
Yes, you are right. I am reading the wrong dimension, have fixed.

Reply:
Why didn't the tests catch this issue?